### PR TITLE
Integrate Slack notifier

### PR DESCRIPTION
**Pull request by nunoalima**
This PR extends the `flathunter` notification service by adding support for [Slack](https://slack.com). 

The Slack notifier can be configured in a similar fashion to the existing notification solutions. Setting up a Slack Incoming Webhook is simple, quick and well documented (https://api.slack.com/messaging/webhooks). It is free to use, reliable and would allow flathunt users to get notified in an application they might already be using at work.

Let me know what you think.

### CODE REVIEW

**Comment by codders** [comment]
Thank you so much for this pull request - looks great! :)

File: flathunter/sender_slack.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    def notify(self, message: str) -> None:
+        """Send message to the Slack webhook"""
+        self.__send_message(message)
+
+    def __send_message(self, message: str) -> None:
+        """Send messages to the Slack webhook"""
+        logger.debug(('webhook_url:', self.webhook_url))
+        logger.debug(('message', message))
+        response = requests.post(
+            self.webhook_url,
```
Comment (codders) [inline flathunter/sender_slack.py:43]: The type-checker doesn't like this line - self.webhook_url may be None:

```
/home/runner/work/flathunter/flathunter/flathunter/sender_slack.py
  /home/runner/work/flathunter/flathunter/flathunter/sender_slack.py:43:13 - error: Argument of type "Unknown | None" cannot be assigned to parameter "url" of type "str | bytes" in function "post"
    Type "Unknown | None" cannot be assigned to type "str | bytes"
      Type "None" cannot be assigned to type "str | bytes"
        Type "None" cannot be assigned to type "str"
        Type "None" cannot be assigned to type "bytes" (reportGeneralTypeIssues)
```

**Review by codders — CHANGES_REQUESTED** [review]
I get a failure in the test_config_wizard suite from the CI. But otherwise I'm very happy with this comprehensive PR - it's got tests, documentation, and updates to the wizard. Amazing work!

File: flathunter/sender_slack.py (hunk 1 above)
Comment (nunoalima) [reply flathunter/sender_slack.py:43]: I've fixed the failing test and also set the default of `self.webhook_url` to an empty string.

Comment (codders) [reply flathunter/sender_slack.py:43]: Looking better. I think the nicer option would have been to abort the 'send_message' function if the webhook_url is None - it's probably also going to raise an error if you try posting data to the URL "". But I'm totally fine with this - keeps the type-checker happy and that seems like an edge case.

**Pull request merged**